browser/ui/webui/new_tab_page/top_sites_message_handler.cc

```diff
@@ -313,27 +313,20 @@ void TopSitesMessageHandler::HandleEditTopSite(const base::Value::List& args) {
   if (title.empty())
     title = new_url.empty() ? url : new_url;
 
+  // when user modifies current top sites, change to favorite mode.
+  if (!most_visited_sites_->IsCustomLinksEnabled()) {
+    profile_->GetPrefs()->SetBoolean(ntp_prefs::kNtpUseMostVisitedTiles, false);
+    most_visited_sites_->EnableCustomLinks(IsCustomLinksEnabled());
+  }
+
   GURL gurl(url);
   GURL new_gurl(new_url);
   std::u16string title16 = base::UTF8ToUTF16(title);
 
-  if (most_visited_sites_->IsCustomLinksEnabled()) {
-    // similar to `MostVisitedHandler::UpdateMostVisitedTile`
-    most_visited_sites_->UpdateCustomLink(
-        gurl, new_gurl != gurl ? new_gurl : GURL(), title16);
-  } else {
-    // when user modifies current top sites, change to favorite mode.
-    profile_->GetPrefs()->SetBoolean(ntp_prefs::kNtpUseMostVisitedTiles, false);
-    most_visited_sites_->EnableCustomLinks(IsCustomLinksEnabled());
-
-    // When user tries to edit from frecency mode, try to edit an existing
-    // site, if one exists. If none exists, add a new custom link for the site.
-    bool updated = most_visited_sites_->UpdateCustomLink(
-        gurl, new_gurl, base::UTF8ToUTF16(title));
-    if (!updated) {
-      most_visited_sites_->AddCustomLink(new_url.empty() ? gurl : new_gurl,
-                                         title16);
-    }
+  bool updated = most_visited_sites_->UpdateCustomLink(gurl, new_gurl, title16);
```

Comment:
super nit: 
```suggestion
  const bool updated = most_visited_sites_->UpdateCustomLink(gurl, new_gurl, title16);
```

Reply:
Cool, thanks for the C++ tip! (mine is decidedly below averaged :laughing:)